crates/network-monitor/src/state.rs

```diff
@@ -0,0 +1,38 @@
+use std::sync::LazyLock;
+
+use tokio::sync::{Mutex, RwLock};
```

Comment:
suggestion: Use either `std::sync::{Mutex, RwLock}` or `parking_lot::{Mutex, RwLock}` (better, but does not change much in this low contention context)

Reply:
Going further, do you really need to use a RwLock?

Reply:
I'm curious about this, the tokio docs for their mutex at once states that it's fine to use the std mutex in an async context, but also states that the std mutex will block threads (which I assume could lead to threadpool exhaustion), and that awaiting while holding the std mutex's lock can easily lead to deadlocks. Should I not prefer correctness over a slight performance improvement?

Reply:
That’s exactly my point, but did I miss a place where you are using it across await points? I don’t think you really need to hold the lock across an await point ever though. Do you think the lock is held for a prolonged period too?

Reply:
It's true, everything done while holding the lock here is fast and there shouldn't be any contention unless something inexplicably spams the set_config endpoint. I was asking because the documentation had surprised me. Shouldn't the first choice be safe code, rather than a tiny performance gain? I can imagine in a more complex example, a developer might add an await point, or add different functions that lock the mutex, and suddenly these problems show up, undetected by the compiler.

Reply:
Alright, I switched to the std lock now